Core/include/Acts/EventData/SpacePointData.hpp

```diff
@@ -39,58 +41,63 @@ class SpacePointData {
+  void setX(const std::size_t idx, const float value);
+  void setY(const std::size_t idx, const float value);
+  void setZ(const std::size_t idx, const float value);
+  void setRadius(const std::size_t idx, const float value);
+  void setPhi(const std::size_t idx, const float value);
+  void setVarianceZ(const std::size_t idx, const float value);
+  void setVarianceR(const std::size_t idx, const float value);
+
+  void setQuality(const std::size_t idx, const float value) const;
+  void setDeltaR(const std::size_t idx, const float value) const;
 
   /// @brief Resize vectors
-  void resize(std::size_t n, bool resizeDynamic = false);
+  void resize(const std::size_t n, bool resizeDynamic = false);
 
   /// @brief clear vectors
   void clear();
 
   ///
-  bool hasDynamicVariable() const { return not m_topStripVector.empty(); }
-
-  const Acts::Vector3& getTopStripVector(std::size_t idx) const {
-    return m_topStripVector[idx];
-  }
-
-  const Acts::Vector3& getBottomStripVector(std::size_t idx) const {
-    return m_bottomStripVector[idx];
-  }
-
-  const Acts::Vector3& getStripCenterDistance(std::size_t idx) const {
-    return m_stripCenterDistance[idx];
-  }
-
-  const Acts::Vector3& getTopStripCenterPosition(std::size_t idx) const {
-    return m_topStripCenterPosition[idx];
-  }
+  bool hasDynamicVariable() const;
 
-  void setTopStripVector(std::size_t idx, const Acts::Vector3& value) {
-    m_topStripVector[idx] = value;
-  }
+  const Acts::Vector3& topStripVector(const std::size_t idx) const;
+  const Acts::Vector3& bottomStripVector(const std::size_t idx) const;
+  const Acts::Vector3& stripCenterDistance(const std::size_t idx) const;
+  const Acts::Vector3& topStripCenterPosition(const std::size_t idx) const;
 
-  void setBottomStripVector(std::size_t idx, const Acts::Vector3& value) {
-    m_bottomStripVector[idx] = value;
-  }
-
-  void setStripCenterDistance(std::size_t idx, const Acts::Vector3& value) {
-    m_stripCenterDistance[idx] = value;
-  }
-
-  void setTopStripCenterPosition(std::size_t idx, const Acts::Vector3& value) {
-    m_topStripCenterPosition[idx] = value;
-  }
+  void setTopStripVector(const std::size_t idx, const Acts::Vector3& value);
+  void setBottomStripVector(const std::size_t idx, const Acts::Vector3& value);
+  void setStripCenterDistance(const std::size_t idx,
+                              const Acts::Vector3& value);
+  void setTopStripCenterPosition(const std::size_t idx,
+                                 const Acts::Vector3& value);
 
  private:
+  /// base variables
+  std::vector<float> m_x{};
+  std::vector<float> m_y{};
+  std::vector<float> m_z{};
+  std::vector<float> m_radius{};
+  std::vector<float> m_phi{};
+  std::vector<float> m_varianceR{};
+  std::vector<float> m_varianceZ{};
+
   /// Mutable variables
-  std::vector<float> m_quality{};
-  std::vector<float> m_deltaR{};
+  mutable std::vector<float> m_quality{};
+  mutable std::vector<float> m_deltaR{};
```

Comment:
Is there no way around `mutable`? One of the main policies was to not have hidden internal mutability.

Reply:
These are quantities that are set during seed finding, and are constantly updated once a new seed is found. Do you have suggestions on an alternative solution here?

Reply:
In my opinion this warrants that they are not in fact held / passed around as `const` and implicitly mutated.